crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -483,7 +529,9 @@ public Response find(FindRequest req) {
         try {
             ctx = newCtx(req, CRUDOperation.FIND);
             ctx.measure.begin("find");
-
+            CompositeMetadata md = ctx.getTopLevelEntityMetadata();
+            metricCtx = metrics.startEntityRequest("find", md.getName(), md.getVersion().getValue());
+            
             StreamingResponse r=_findAndStream(req,ctx);
```

Comment:
We can handle streaming in the mediator, not HTTP (lightblue-rest) layer.

We want to hook into when the document stream is fully consumed or closed. Closed might be the best indicator.

One simple way I can think of to do this is to write a MeteredDocumentStream that delegates to another stream entirely with the addition that when the close() method is called, the MeteredDocumentStream __also__ ends a metric ctx.

Then in `_findAndStream`, instead of:

```
            CRUDFindResponse result = finder.find(ctx, req.getCRUDFindRequest());
            ctx.measure.end("finder.find");
            
            if(!ctx.hasErrors()) {
                ctx.setStatus(OperationStatus.COMPLETE);
                response.documentStream=ctx.getDocumentStream();
                if(ctx.isComputeCounts()) {
                    response.matchCount=result.getSize();
                }
            } else {
                ctx.setStatus(OperationStatus.ERROR);
            }
```

...you'd do something like (accepting a `RequestMetric.Context metricCtx` as a method argument):

```
            CRUDFindResponse result = finder.find(ctx, req.getCRUDFindRequest());
            ctx.measure.end("finder.find");
            
            if(!ctx.hasErrors()) {
                ctx.setStatus(OperationStatus.COMPLETE);
                response.documentStream=new MeteredDocumentStream(ctx.getDocumentStream(), metricCtx);
                if(ctx.isComputeCounts()) {
                    response.matchCount=result.getSize();
                }
            } else {
                ctx.setStatus(OperationStatus.ERROR);
                // Do we need to do this in more places?
                ctx.getErrors().forEach(metricCtx::markRequestException);
                metricCtx.endRequestMonitoring();
            }
```